2.6_chiseltest.ipynb

```diff
@@ -503,6 +503,36 @@
     "}\n"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "// test for backpressure\n",
+    "test(new DecoupledGcd(16)) { dut =>\n",
+    "  dut.input.initSource().setSourceClock(dut.clock)\n",
+    "  dut.output.initSink().setSinkClock(dut.clock)\n",
+    "\n",
+    "  val testValues = for { x <- 1 to 10; y <- 1 to 10} yield (x, y)\n",
+    "  val inputSeq = testValues.map { case (x, y) =>\n",
+    "    (new GcdInputBundle(16)).Lit(_.value1 -> x.U, _.value2 -> y.U)\n",
+    "  }\n",
+    "  val resultSeq = testValues.map { case (x, y) =>\n",
+    "    new GcdOutputBundle(16).Lit(_.value1 -> x.U, _.value2 -> y.U, _.gcd -> BigInt(x).gcd(BigInt(y)).U)\n",
+    "  }\n",
+    "\n",
+    "  for((inputVal, resultVal) <- inputSeq zip resultSeq) {\n",
+    "    dut.input.enqueueNow(inputVal)\n",
+    "    dut.output.ready.poke(false.B)\n",
+    "    for(_ <- 0 to 10) {\n",
+    "      dut.clock.step(1)\n",
```

Comment:
this is my poor mans solution for the clock divider - set the ouput in a non-ready state and idle for come cycles.